jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/CliDownloaderUtil.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.http.HttpClient;
+import io.fabric8.kubernetes.client.http.HttpResponse;
+import io.fabric8.kubernetes.client.utils.HttpClientUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.commons.io.FilenameUtils.removeExtension;
+import static org.eclipse.jkube.kit.common.archive.ArchiveDecompressor.extractArchive;
+
+public class CliDownloaderUtil {
+  private CliDownloaderUtil() { }
+
+  public static String downloadCli(String baseDownloadUrl, String cliBinaryPrefix, String cliArtifactName, File outputDirectory) throws IOException {
```

Comment:
I'm still not sure why we need to compute the final URI from different components.

Also, I don't understand why the return type is a String instead of a Path or a File.

Some Javadoc to document the method would be nice too.